DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/EnrolController.cs

```diff
@@ -257,11 +254,7 @@ public IActionResult EnrolDelegateSupervisor(int delegateId, EnrolSupervisorView
                 sessionEnrol.SupervisorName = supervisorList.FirstOrDefault(x => x.AdminId == model.SelectedSupervisor).Name;
                 sessionEnrol.SupervisorID = model.SelectedSupervisor;
             }
-            if (model.SelectedSupervisorRoleId.HasValue && model.SelectedSupervisorRoleId.Value > 0)
-            {
-                sessionEnrol.SelfAssessmentSupervisorRoleName = roles.FirstOrDefault(x => x.ID == model.SelectedSupervisorRoleId).RoleName;
-            }
-            sessionEnrol.SelfAssessmentSupervisorRoleId = model.SelectedSupervisorRoleId;
+            sessionEnrol.SelfAssessmentSupervisorRoleId = roles.FirstOrDefault().ID;

```

Comment:
This makes the assumption that there will only be one role returned by the supervisorService (or that the first one should be chosen). While this will work for the existing self assessments, it won't work if a self assessment has more than one role that can be assigned to a supervisor. I think we still need to handle the scenario where there might be more than 1.